Email verification notice

hummus-erectus opened the pull request:
Instead of automatically navigating to the sign-in page after a user signs up, a message is displayed asking them to check their email to verify their account. They can click a button to navigate to the sign-in page.

Users will also now see speciic error messages on the sign-in page if they have not verified their email or if the user credentials are incorrect.

hummus-erectus committed:
Update useAuthStore to give specific error messages

hummus-erectus committed:
Add email verification message and sign in button upon sign up

hummus-erectus committed:
Hide inputs on sign up form upon succesful sign up

jjmountain commented on app/sign-up.tsx:
```diff
@@ -24,9 +25,15 @@ export default function SignUpPage() {
+      }
+    } catch (error) {
+      // Handle any sign-up errors here
+      console.error('Sign-up error:', error);
```
I think it would be a good idea to show this text so the user can see if there's an error (in addition to logging to the console)

hummus-erectus replied:
Error handling is already done in the useAuthStore's signUp function, which sets the error state, and that is displayed to the screen 

`              {error && <Text className="text-red-500 mt-2 text-center">{error}</Text>}
`

I'll remove this console.log and try-catch block, if we need the logs for errors I think it might be best to handle them in useAuthStore?

jjmountain commented on app/sign-up.tsx:
```diff
@@ -92,19 +99,31 @@ export default function SignUpPage() {
+        {signUpMessage ? (
+          <View className="mb-4">
+            <Text className="text-blue-500 font-bold font-lg text-center mb-2">{signUpMessage}</Text>
+            <TouchableOpacity
```
I saw a warning in the console today that TouchableOpacity is going to be deprecated and replaced by Pressable in the future. Let's use this instead, unless there's a good reason we shouldn't (I'm aware we'll have a lot of them to change in the codebase as Claude chose to use them).

see https://reactnative.dev/docs/touchableopacity

hummus-erectus replied:
Yeah I've seen a few similar warnings about deprecated elements, shall I work through all of them?

jjmountain commented on hooks/useAuthStore.ts:
```diff
@@ -36,11 +36,20 @@ export const useAuthStore = create<AuthState>((set) => ({
+        console.log("Signin error:", error.message)
+        let errorMessage = 'Failed to sign in';
+        if (error.message.includes('Invalid login credentials')) {
+          errorMessage = 'Invalid email or password. Please try again.';
```
Hmm I'm guessing you were trying to change the error message here for the user's benefit. I think this is something we can revisit and for now it's best we see what the actual `error.message` is from Supabase in the UI.

We should be throwing / setting the error with `error.message`

It would be nice if we could surface this error in the UI on the sign in page (also for sign up)

hummus-erectus replied:
Yes that was my thinking. So for now I will simplify and use the direct message as you suggested.

jjmountain commented on hooks/useAuthStore.ts:
```diff
@@ -49,10 +58,12 @@ export const useAuthStore = create<AuthState>((set) => ({
     } catch (error) {
       console.error('Error signing up:', error);
       set({ isLoading: false, error: 'Failed to sign up' });
+      throw error;
```
Why are we throwing this error here? The reason for throwing an error is to bail out of a method and hit the next catch block but this is doing nothing useful.

jjmountain replied:
Probably what we should do here for now is set some variables signUpError and signInError to the two different errors and try to surface them in the code.

hummus-erectus replied:
I think this was a Claude suggestion as some kind of failsafe when I asked it to check my code. I'll be more careful not to take it at it's word!

jjmountain commented:
Thanks for taking the initiative on this one - it's a good improvement. 

As you're changing some of the error logic I've put some suggestions in around how we can handle / show errors which would make sense to fix now.

hummus-erectus committed:
Refactor error logic for sign-up

hummus-erectus commented:
I have just added 2 commits.

The first is one I worked on last night and removes the form elements when the user successfully completes sign up.

The second addresses the issues in the previous commit. All displayed error messages are directly those provided by Supabase. Sign up success is now managed in useAuthStore as a booelan and the message is set in the sign-up component.

hummus-erectus committed:
Remove unnecessary returns in useAuthStore.ts signUp function

hummus-erectus committed:
Restore sign-up to current convention regarding button elements

jjmountain commented:
There's a lot in this commit now that shouldn't be there and actually would revert the changes that I made. I'm not sure what you did, but I think maybe you have a fundamental misunderstanding that you are supposed to be keeping your branches up to date with master (it seems that way as you're trying to include commits from elsewhere at least). 

You don't need to do this though. The key is to look in files changed tab and see which are changed - you should only see the changes you are trying to add which are signalled in the pull request description. Any others and it means you've messed  something up! Hope that is helpful. I'll fix this now by removing the commits that aren't yours.

jjmountain commented:
Can you look in files changed and confirm for me that this is what you intended?

hummus-erectus reviewed (commented):
Yes this is the intended update!

jjmountain commented:
Let's get this merged then! 😀

hummus-erectus reviewed (commented):
Good to go

jjmountain commented:
lol you need to merge it 😆
